src/server/handler/mod.rs

```diff
@@ -55,7 +55,7 @@ impl From<Arc<Config>> for HttpHandler {
             };
         }
 
-        let file_server = FileServer::new(config.root_dir());
+        let file_server = FileServer::new(config.root_dir(), config.clone());
```

Comment:
Why not just borrowing `config`?

```suggestion
        let file_server = FileServer::new(&config);
```